apps/extension/src/ui/apps/dashboard/routes/Settings/Accounts/NewFolderModal.tsx

```diff
@@ -0,0 +1,170 @@
+import { yupResolver } from "@hookform/resolvers/yup"
+import Dialog from "@talisman/components/Dialog"
+import { Modal } from "@talisman/components/Modal"
+import { ModalDialog } from "@talisman/components/ModalDialog"
+import { useOpenClose } from "@talisman/hooks/useOpenClose"
+import { provideContext } from "@talisman/util/provideContext"
+import { api } from "@ui/api"
+import useAccountsCatalog from "@ui/hooks/useAccountsCatalog"
+import { RefCallback, useCallback, useEffect, useMemo, useRef } from "react"
+import { useForm } from "react-hook-form"
+import { useTranslation } from "react-i18next"
+import styled from "styled-components"
+import { Checkbox, FormFieldContainer, FormFieldInputText } from "talisman-ui"
+import * as yup from "yup"
+
+const useNewFolderModalProvider = () => {
+  const { isOpen, open, close } = useOpenClose()
+
+  useEffect(() => {
+    close()
+  }, [close])
+
+  return {
+    isOpen,
+    open,
+    close,
+  }
+}
+
+export const [NewFolderModalProvider, useNewFolderModal] = provideContext(useNewFolderModalProvider)
+
+export const NewFolderModal = () => {
+  const { t } = useTranslation("admin")
+  const { close, isOpen } = useNewFolderModal()
+
+  return (
+    <Modal open={isOpen}>
+      <ModalDialog title={t("New Folder")} onClose={close}>
+        <NewFolder onConfirm={close} onCancel={close} />
+      </ModalDialog>
+    </Modal>
+  )
+}
+
+const StyledDialog = styled(Dialog)`
+  .error {
+    font-size: var(--font-size-small);
+    color: var(--color-status-warning);
+    height: 1.6em;
+    margin-bottom: -1.6em;
+  }
+`
+
+type FormData = {
+  name: string
+  followedOnly?: boolean
+}
+
+interface NewFolderProps {
+  onConfirm: () => void
+  onCancel: () => void
+  className?: string
+}
+
+const NewFolder = ({ onConfirm, onCancel, className }: NewFolderProps) => {
+  const { t } = useTranslation("admin")
+
+  const schema = useMemo(
+    () =>
+      yup
+        .object({
+          name: yup.string().required(""),
```

Comment:
Might want to implement a "A folder with this name already exist" here
Tried this scenario, there's no error displayed but it doesn't work (no new folder created when submitting).